src/components/AddComment.jsx

```diff
@@ -0,0 +1,64 @@
+import React, { useState } from "react";
+import { addComment } from "../api";
+
+export const AddComment = ({ article_id, addCommentToList }) => {
+  const [usernameInput, setUsernameInput] = useState("");
+  const [commentInput, setCommentInput] = useState("");
+  const [isSubmitting, setIsSubmitting] = useState(false);
+  const [successMessage, setSuccessMessage] = useState("");
+
+  const handleSubmit = (event) => {
+    event.preventDefault();
+
+    if (!usernameInput || !commentInput) {
+      alert("Input cannot be empty.");
+      return;
+    }
+
+    setIsSubmitting(true);
+    addComment(article_id, {username: usernameInput, body: commentInput})
+      .then((newComment) => {
+        addCommentToList(newComment.comment);
+        setSuccessMessage("Comment posted successfully!");
+        setCommentInput("");
+        setUsernameInput("");
+        setIsSubmitting(false)
+      })
+      .catch((error) => {
+    alert(`Error: {${error.response ? error.response.data.msg : error.message}}`);
+        setIsSubmitting(false);
+      });
+  };
+
+  return (
+    <section className="post-comment">
+      <h3>Add a Comment</h3>
+      <form onSubmit={handleSubmit}>
+        <label htmlFor="usernameInput">Username:</label>
+        <input
+           type="text"
+          id="usernameInput"
+          value={usernameInput}
+          onChange={(e) => setUsernameInput(e.target.value)}
+          placeholder="Enter your username"
+          disabled={isSubmitting}
+          required
+        />
```

Comment:
Instead of a username field you can optionally hard code this value for now. Currently as a user I have to know a valid username to be able to post a comment.